python/src/ecole/environment.py

```diff
@@ -109,16 +109,21 @@ def reset(self, instance, *dynamics_args, **dynamics_kwargs):
 
             self.dynamics.set_dynamics_random_state(self.model, self.random_engine)
 
-            self.observation_function.before_reset(self.model)
             self.reward_function.before_reset(self.model)
+            self.observation_function.before_reset(self.model)
             self.information_function.before_reset(self.model)
             done, action_set = self.dynamics.reset_dynamics(
                 self.model, *dynamics_args, **dynamics_kwargs
             )
 
-            observation = self.observation_function.extract(self.model, done)
+            if not done:
+                observation = self.observation_function.extract(self.model, done)
+            else:
+                observation = None
             reward_offset = self.reward_function.extract(self.model, done)
```

Comment:
Wasn't `reward_offset` supposed to be collected before the observation now? Maybe I messed up something when rebasing.